packages/app/src/specs/generators/component/ReactComponentGeneratorStepOne.vue

```diff
@@ -0,0 +1,198 @@
+
+const props = defineProps<{
+  title: string
+  gql: ReactComponentGeneratorStepOne_CodeGenGlobFragment
+}>()
+const { t } = useI18n()
+const emits = defineEmits<{
+  (event: 'update:title', value: string): void
+  (event: 'update:description', value: string): void
+  (event: 'restart'): void
+  (event: 'close'): void
+}>()
+const { title } = useVModels(props, emits)
+
+title.value = t('createSpec.component.importFromComponent.chooseAComponentHeader')
+gql`
+fragment ReactComponentGeneratorStepOne_codeGenGlob on CurrentProject {
+  id
+  codeGenGlobs {
+    id
+    component
+  }
+  codeGenFramework
+}
+`
+
+gql`
+query ReactComponentGeneratorStepOne($glob: String!) {
+  currentProject {
+    id
+    codeGenCandidates(glob: $glob) {
+      id
+      fileName
+      fileExtension
+      absolute
+      relative
+      baseName
+    }
+    # Add the specs, so we can keep the list up to date with the cache
+    specs {
+      id
+      ...SpecNode_InlineSpecList
+    }
+  }
+}
+`
+
+gql`
+mutation ReactComponentGeneratorStepOne_generateSpec($codeGenCandidate: String!, $type: CodeGenType!, $componentName: String!) {
+  generateSpecFromSource(codeGenCandidate: $codeGenCandidate, type: $type, componentName: $componentName) {
+    ...GeneratorSuccess
+    currentProject {
+      id
+      ...EmptyGenerator
+    }
+    generatedSpecResult {
+      ... on GeneratedSpecError {
+        fileName
+      }
+    }
+  }
+}`
+
+const generateSpecMutation = useMutation(ReactComponentGeneratorStepOne_GenerateSpecDocument)
+const extensionPattern = ref(props.gql.codeGenGlobs.component)
+
+const query = useQuery({
+  query: ReactComponentGeneratorStepOneDocument,
+  // @ts-ignore
+  variables: { glob: extensionPattern },
+})
+const allFiles = computed((): any => {
+  if (query.data.value?.currentProject?.codeGenCandidates) {
+    return query.data.value.currentProject?.codeGenCandidates
+  }
+
+  return []
+})
+const result = ref<GeneratorSuccessFileFragment | null>(null)
+const generatedSpecError = ref()
+const generateSpecFromSource = ref()
+
+whenever(result, () => {
+  title.value = t('createSpec.successPage.header')
+})
+
+whenever(generatedSpecError, () => {
+  title.value = t('createSpec.component.importEmptySpec.header')
+})
+
+const makeSpec = async ({ file, item }) => {
+  const { data } = await generateSpecMutation.executeMutation({
+    codeGenCandidate: file.absolute,
+    type: 'component',
+    componentName: item,
+  })
+
+  generateSpecFromSource.value = data?.generateSpecFromSource
+  result.value = data?.generateSpecFromSource?.generatedSpecResult?.__typename === 'ScaffoldedFile' ? data?.generateSpecFromSource?.generatedSpecResult : null
+  generatedSpecError.value = data?.generateSpecFromSource?.generatedSpecResult?.__typename === 'GeneratedSpecError' ? data?.generateSpecFromSource?.generatedSpecResult : null
+}
+const cancelSpecNameCreation = () => {
+  generatedSpecError.value = null
+}
+</script>
+<style scoped>
+.ghost-div {
```

Comment:
Another good reason to move from Windi to Tailwind... we can use arbitrary values https://v2.tailwindcss.com/docs/just-in-time-mode#arbitrary-value-support. You can just do `class=[calc(100% - 24px)]` using their arbitrary value feature.

I don't think this works here right now, but I hope we can move to "real" Tailwind soon. The more I use it over Windi, the more I want to migrate.

Reply:
We can do this kind of thing already, see the `w-[calc(100%-2px)]` [here](https://github.com/cypress-io/cypress/blob/develop/packages/app/src/specs/InlineSpecList.vue#L21).

But still I agree with switching to real tailwind soon for the other reasons you've mentioned.

Reply:
Oh, neat - we can make the refactor here, then!

Still look forward to mainline Tailwind. Windi has some neat features, and there's a time and place for trying new libraries, but a production app like Cypress isn't one of them.